### PR TITLE
232/csq consequence weighting sampling

### PR DESCRIPTION
**Pull request by torbjorgen**

### CODE REVIEW

**Comment by torbjorgen** [comment]
- [x] Check run /home/tor.bjorgen/repos/cg/next-test-train 8711304

**Comment by torbjorgen** [comment]
Check job 8714969, /home/tor.bjorgen/repos/cg/next-test-train e3f8b14

**Comment by torbjorgen** [comment]
Check job 8714970 /home/tor.bjorgen/repos/cg/raredisease-ml 414c4ef

**Comment by torbjorgen** [comment]
Check job Submitted batch job 8723424
/home/tor.bjorgen/repos/cg/raredisease-ml
7e265b2

**Comment by torbjorgen** [comment]
Check job 8723426 78e9464 /home/tor.bjorgen/repos/cg/next-test-train

image

Slight performance increase. Increase weight going forward.

**Comment by torbjorgen** [comment]
Check job 78e9464 8725520 /home/tor.bjorgen/repos/cg/raredisease-ml

**Comment by torbjorgen** [comment]
Dataset contains majority clinvar annotated BENIGN variants (600k), clinvar annotated pathogenic is 90k.

# Train
Model has a high F1 and MCC score on training data, i.e low FP and FN rate.
Binary accuracy on train is high.

Performance is good on common pathogenic variants(1m).

Model is performing poorly on rare variants (500k) and variants without clinvar annotation (100k).

Also performing poorly on rare variants (as above) on subset without CLINVAR annotations.

Model is also performing poorly on various VEP sub categories (~10k samples each) except for
- missense variants, F1 0.42 (150k)
- stop gained F1 0.41 (48k)
- frameshift variant F1 0.58 (80k)

interestingly, intronic variants are the most common (2m) and F1 is poor F1 0.12
but it can be because of low TPs in this category.

# Test
Test dataset is poor (overfitting issue?) on all VEP categories and F1, MCC.

However model correctly rejects a lot of TNs and has a low FP rate.

# Summary
Difference CLINVAR and MUTACC is that MUTACC only contains rare variants
which is shown to be poorly recognized (see above summary).

# TODOs
- Consider using only pathogenic variants in VEP metrics? Or only use COUNT for TPs?
This will bring more insightful metrics, since they're now obscured by a lot of
negative benign samples. On the other hand, we we do want the FP count per category,
so just selcting TPs for F1 score metric is biasing results.

- Weight rare variants higer while training

**Comment by torbjorgen** [comment]
This PR is deprecated in favor of changes in branch 2291e35 `(origin/performantmodel-rc0, performantmodel-rc0) Earlystop on train loss`.
